src/Components/Components/src/ComponentsMetrics.cs

```diff
@@ -0,0 +1,239 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Components;
+
+internal sealed class ComponentsMetrics : IDisposable
+{
+    public const string MeterName = "Microsoft.AspNetCore.Components";
+    public const string LifecycleMeterName = "Microsoft.AspNetCore.Components.Lifecycle";
+    private readonly Meter _meter;
+    private readonly Meter _lifeCycleMeter;
+
+    private readonly Counter<long> _navigationCount;
+
+    private readonly Histogram<double> _eventDuration;
+    private readonly Counter<long> _eventException;
+
+    private readonly Histogram<double> _parametersDuration;
+    private readonly Counter<long> _parametersException;
+
+    private readonly Histogram<double> _batchDuration;
+    private readonly Counter<long> _batchException;
+
+    public bool IsNavigationEnabled => _navigationCount.Enabled;
+
+    public bool IsEventDurationEnabled => _eventDuration.Enabled;
+    public bool IsEventExceptionEnabled => _eventException.Enabled;
+
+    public bool IsParametersDurationEnabled => _parametersDuration.Enabled;
+    public bool IsParametersExceptionEnabled => _parametersException.Enabled;
+
+    public bool IsBatchDurationEnabled => _batchDuration.Enabled;
+    public bool IsBatchExceptionEnabled => _batchException.Enabled;
+
+    public ComponentsMetrics(IMeterFactory meterFactory)
+    {
+        Debug.Assert(meterFactory != null);
+
+        _meter = meterFactory.Create(MeterName);
+        _lifeCycleMeter = meterFactory.Create(LifecycleMeterName);
+
+        _navigationCount = _meter.CreateCounter<long>(
+            "aspnetcore.components.navigation",
+            unit: "{route}",
+            description: "Total number of route changes.");
+
+        _eventDuration = _meter.CreateHistogram(
+            "aspnetcore.components.event.duration",
+            unit: "s",
+            description: "Duration of processing browser event.",
+            advice: new InstrumentAdvice<double> { HistogramBucketBoundaries = MetricsConstants.ShortSecondsBucketBoundaries });
+
+        _eventException = _meter.CreateCounter<long>(
+            "aspnetcore.components.event.exceptions",
+            unit: "{exception}",
+            description: "Total number of exceptions during browser event processing.");
+
+        _parametersDuration = _lifeCycleMeter.CreateHistogram(
+            "aspnetcore.components.update_parameters.duration",
+            unit: "s",
+            description: "Duration of processing component parameters.",
+            advice: new InstrumentAdvice<double> { HistogramBucketBoundaries = MetricsConstants.ShortSecondsBucketBoundaries });
```

Comment:
`ShortSecondsBucketBoundaries` is designed for network level latencies such as an http request. I suspect that components may need buckets more focused on the sub ms level timings?

Reply:
This could involve network requests or not. 
Added `BlazorPropertyUpdateSecondsBucketBoundaries = [0.000001, 0.00001, 0.0001, 0.001, 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10]`